From 07e5a3ac2858e6d38cc698e0f740e7a693e9f302 Mon Sep 17 00:00:00 2001
From: "Friedrich W. H. Kossebau" <kossebau@kde.org>
Date: Mon, 16 Nov 2020 17:14:52 +0100
Subject: [PATCH] Make plugins compatible to latest KPluginFactory code

That one uses std::is_constructible<impl, QObject *, const QVariantList &>
which will fail for our current plugin constructors due to being private
and with only friend class KPluginFactory.
---
 src/plugins/dummy/dummybackend.h   | 4 +---
 src/plugins/sfdisk/sfdiskbackend.h | 4 +---
 2 files changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/plugins/dummy/dummybackend.h b/src/plugins/dummy/dummybackend.h
index 815b020..8d67158 100644
--- a/src/plugins/dummy/dummybackend.h
+++ b/src/plugins/dummy/dummybackend.h
@@ -24,11 +24,9 @@ class QString;
 */
 class DummyBackend : public CoreBackend
 {
-    friend class KPluginFactory;
-
     Q_DISABLE_COPY(DummyBackend)
 
-private:
+public:
     DummyBackend(QObject* parent, const QList<QVariant>& args);
 
 public:
diff --git a/src/plugins/sfdisk/sfdiskbackend.h b/src/plugins/sfdisk/sfdiskbackend.h
index e709f61..0571add 100644
--- a/src/plugins/sfdisk/sfdiskbackend.h
+++ b/src/plugins/sfdisk/sfdiskbackend.h
@@ -27,11 +27,9 @@ class QString;
 */
 class SfdiskBackend : public CoreBackend
 {
-    friend class KPluginFactory;
-
     Q_DISABLE_COPY(SfdiskBackend)
 
-private:
+public:
     SfdiskBackend(QObject* parent, const QList<QVariant>& args);
 
 public:
-- 
GitLab

